### ISSUE TITLE
FEAT: endpoint to list all graphs

### ISSUE THREAD

**Issue by edwinkys** [oasysai/oasysdb#15]
### Use case

This feature will allow the users to list all of the graphs created by calling `GET /graphs`. This is helpful if users want to delete some graphs, this feature will give them clarity on which graph to delete and why.

### Proposed solution

When the users call `GET /graphs`, this endpoint should return an object of the graphs name and the graph configuration. It should not return the graph itself since it's not useful for any use case. Please see the object below for some reference.

```json
{
  "my_graph": {
    "name": "my_graph",
    "ef_construction": 100,
    "ef_search": 100,
    "filter": {
      "type": "text"
      "username": "edwinkys"
    }
  }
}
```

**Comment by Ashad-h** [comment]
Why not adding the size of the graph (number of keys) ?

Like this:

```json
{
  "my_graph": {
    "name": "my_graph",
    "ef_construction": 100,
    "ef_search": 100,
    "filter": {
      "type": "text"
      "username": "edwinkys"
    },
    "size": 100
  }
}
```

**Comment by edwinkys** [comment]
@Ashad-h That sounds good. We can add that detail to the ‘GraphConfig’ struct and during the graph building function.

**Comment by Ashad-h** [comment]
Also we have to find a way to not deserialize the whole graph because right now, it's mandatory to do it to fetch the config of the graph.

**Comment by edwinkys** [comment]
So, you're saying that if we have a lot of graphs, this endpoint will cause severe performance issue since we have to deserialize the graph too even though we only need the config?

**Comment by Ashad-h** [comment]
Yes, because of this line [https://github.com/oasysai/oasysdb/blob/6c73670878767285b0214427f562ca3b7009f2a4/src/db/database.rs#L198C20-L198C20](url)
You would have to do this to get the graph config. But I think this loads the entire graph in memory from the disk so It would slow down since you would have to do it for every stored graph (only when not being in in-memory database mode obviously)

**Comment by edwinkys** [comment]
Hmm. Do you think it would make sense to store the graph and the graph config separately? So that for this endpoint, we can just get the graph configs without the graph itself.
